Os 297 system entities

jared-improbable opened the pull request.

jared-improbable committed:
Added display name and player id to locator options
Fixed memory leak of futures
`UseInsecureConnection = true`

jared-improbable committed:
Implement members

jared-improbable committed:
Simplified internal locator connection api
Fixed protocol logging options not propagating in Locator flow

jared-improbable committed:
Better names
Properly plumb through UseInsecureConnection

jared-improbable committed:
Need the port too

jared-improbable committed:
Remove `AssociatePathWithClient` command

jared-improbable committed:
Default to receptionist-only connection flow
Remove `LogMessageOp` handling

jared-improbable committed:
Use worker type rather than worker attributes for write access

jared-improbable committed:
Removed `GetMetrics` command

jared-improbable committed:
Make `dev-token` an optional argument

jared-improbable committed:
Compile fix

jared-improbable commented on Workers/GameLogic/Program.cs:
```diff
@@ -30,9 +28,8 @@ private static async Task<int> Main(string[] args)
-            Parser.Default.ParseArguments<ReceptionistOptions, LocatorOptions>(args)
-                .WithParsed<ReceptionistOptions>(opts => options = opts)
-                .WithParsed<LocatorOptions>(opts => options = opts);
+            Parser.Default.ParseArguments<ReceptionistOptions>(args)
```
Bias to writing managed workers rather than client workers.

jared-improbable commented on Workers/GameLogic/Program.cs:
```diff
@@ -105,32 +102,6 @@ private static async Task RunAsync(IWorkerOptions options)
 
         private static void ProcessOpList(OpList opList)
         {
-            foreach (var logOp in opList.OfOpType<LogMessageOp>())
```
They're deprecated anyway, just kill 'em

jamiebrynes7 replied:
Wait `LogMessageOp` is deprecated! Brb going to remove this in https://github.com/spatialos/gdk-for-unity too

jared-improbable commented on Improbable/Stdlib/Improbable.Stdlib/WorkerConnection.cs:
```diff
@@ -186,45 +187,63 @@ public void StopSendingMetrics()
+        private static string GetDevelopmentPlayerIdentityToken(string host, ushort port, bool useInsecureConnection, string authToken, string playerId, string displayName)
         {
-            var pit = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
+            using (var pit = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
```
Memory leak! cc @francissauch

francissauch replied:
my bad! good catch :)

jared-improbable committed:
Use built-in env var

jared-improbable committed:
Dbg

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/WorkerConnection.cs:
```diff
@@ -61,7 +60,7 @@ public static Task<WorkerConnection> ConnectAsync(IWorkerOptions workerOptions,
                 case ILocatorOptions locatorOptions:
                     connectionParameters.Network.UseExternalIp = true;
-                    return ConnectAsync(locatorOptions.SpatialOsHost, locatorOptions.SpatialOsPort, connectionParameters, locatorOptions.Token, "Player", "", connectionParameters.WorkerType, taskOptions);
+                    return ConnectAsync(locatorOptions.SpatialOsHost, locatorOptions.SpatialOsPort, locatorOptions.UseInsecureConnection, connectionParameters, locatorOptions.DevToken, locatorOptions.PlayerId, locatorOptions.DisplayName, taskOptions);
```
Any reason the `ConnectAsync` overload doesn't take `ILocatorOptions`?

jamiebrynes7 replied:
Bump 😄

jared-improbable replied:
Old API - have updated!

jamiebrynes7 reviewed (approved):
LGTM

jared-improbable committed:
Fixed SQL to work with PSQL below 11.x

jared-improbable committed:
Add a simple Platform SDK wrapper
Managed `spatiald`, and simplifies starting and stopping local dpls
Added concrete (but incomplete) types for local dpl and spatialos.json files

jared-improbable committed:
Version bump

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/DeploymentWorkerFlags.cs:
```diff
@@ -0,0 +1,28 @@
+
+            WorkerType = workerType;
+            Flags = ImmutableArray<Flag>.Empty;
+            Flags = Flags.AddRange(flags);
```
Can we not initialize this directly? 😱

jared-improbable replied:
Nope, the API is a bit...yuck.

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/LayerConfiguration.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace Improbable.Stdlib.Platform
+{
+    public readonly struct LayerConfiguration
```
Worth adding a small note that this is incomplete?

jared-improbable replied:
Done

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+                "spatial", "alpha", "config", "convert", "--json_output", "--launch_config", Shell.Escape(startDeployment.LaunchConfigPath),
+                "--main_config", Shell.Escape(startDeployment.ProjectConfigPath), "--output_path", splConfigPath);
+
+            var project = JsonConvert.DeserializeObject<ProjectConfig>(File.ReadAllText(startDeployment.ProjectConfigPath, Encoding.UTF8));
```
Probably worth try-catching to handle breaking changes in the FPL config files?

jared-improbable replied:
I'd just let exceptions bubble out

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+            var splConfigPath = Path.Combine(tempDir, "spl_config.json");
+            Shell.Run(
+                "spatial", "alpha", "config", "convert", "--json_output", "--launch_config", Shell.Escape(startDeployment.LaunchConfigPath),
+                "--main_config", Shell.Escape(startDeployment.ProjectConfigPath), "--output_path", splConfigPath);
```
This is not ideal 😕

jared-improbable replied:
Nuh uh. FPL needs finishing off.

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+            {
+                try
+                {
+                    Process.GetProcessById(pid);
```
What if `spatiald` has been started with a different project?

jared-improbable replied:
I'm not worried about it for now.

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+                var deployments = client.ListDeployments(new ListDeploymentsRequest
+                    { DeploymentName = "local", ProjectName = project.ProjectName }, timeout);
+
+                foreach (var dpl in deployments.Where(dpl => dpl.Status == Deployment.Types.Status.Running))
```
You should be able to add a filter onto the `ListDeploymentsRequest`? Or at least you can when using `spot` (which leads to me believe you can with the platform SDK

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+                var deployments = client.ListDeployments(new ListDeploymentsRequest
+                    { DeploymentName = "local", ProjectName = project.ProjectName }, timeout);
+
+                foreach (var dpl in deployments.Where(dpl => dpl.Status == Deployment.Types.Status.Running))
```
What happens if the deployments call hits the time out?

jared-improbable replied:
`TaskCancelledException` is thrown

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+            return new PlatformApiChannel(GetCredentials(), new PlatformApiEndpoint("localhost", 9876, true));
+        }
+
+        public static Deployment StartLocal(in StartDeploymentOptions startDeployment)
```
Any reason you aren't using async-await here? There should be async functions for the Platform SDK calls

jared-improbable replied:
Nope, just started as a direct port of the inline versions I was using the now-dead SpotSharp CLI tool

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+                var allWorkerFlags = new RepeatedField<WorkerFlag>();
+
+                var startFlags = startDeployment.WorkerFlags.SelectMany(wf =>
+                    wf.Flags.Select(f => new WorkerFlag { Key = f.Name, Value = f.Value, WorkerType = wf.WorkerType }));
```
Worth providing interop between on the `Platform.DeploymentWorkerFlags` object and this one to tidy this up? 

I.e. - 

```csharp
public IEnumerable<WorkerFlag> ToPlatformSdk()
{
	...
}
```

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+            return response;
+        }
+
+        public static void Stop(in ProjectConfig config)
```
```suggestion
        public static void StopAllDeployments(in ProjectConfig config)
```

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+                Shell.Run("spatial", "service", "start", "--json_output", "--main_config", Shell.Escape(startDeployment.ProjectConfigPath));
+            }
+
+            using (var tcs = new CancellationTokenSource(5000))
```
Can use the `Stop(...)` function defined below 😄

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+                    Description = config.ProjectName,
+                    Lifetime = Duration.FromTimeSpan(TimeSpan.FromMinutes(30)),
+                    ProjectName = config.ProjectName
+                }, timeout);
```
What happens if we hit the timeout?

jared-improbable replied:
`TaskCancelledException` is thrown

jamiebrynes7 commented on Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs:
```diff
@@ -0,0 +1,236 @@
+
+            public static string Escape(string arg)
+            {
+                return arg.Any(char.IsSeparator) ? $"\"{arg}\"" : arg;
```
💯

jamiebrynes7 commented on ci/test.sh:
```diff
@@ -4,6 +4,8 @@ set -euxo pipefail
 
 cd "$(dirname "$0")/../"
 
+ls -lah $IMPROBABLE_CONFIG_DIR/oauth2
```
Probably remove this now?

jamiebrynes7 reviewed (commented):
Need a linting pass over the `Platform/Platform.cs` file I think 😄

jared-improbable committed:
Format

jared-improbable committed:
Cleanup

jared-improbable committed:
CR Feedback

jared-improbable committed:
Code cleanup
Fixed disconnection not cancelling all outstanding command requests

jared-improbable committed:
Fixed being able to deadlock by `await`ing a command on a connection which has been disconnected

jared-improbable committed:
Proper async implmentations with progress and cancellation
Code cleanup

jared-improbable committed:
Error handling, more debug info

jared-improbable committed:
Process handle leak

jared-improbable committed:
Bit of debugging

jared-improbable committed:
Latest toolbelt

jared-improbable committed:
Workflows feedback: Try a sleep

jared-improbable committed:
Tracking ticket

jared-improbable committed:
Typo fixes

jared-improbable committed:
Version bumps

jared-improbable committed:
Version bump

jared-improbable committed:
Version
